pkg/models/application.go

```diff
@@ -0,0 +1,27 @@
+package models
+
+type AutomationType string
+
+const (
+	AutomationTypeHelm      AutomationType = "helm"
+	AutomationTypeKustomize AutomationType = "kustomize"
+)
+
+type SourceType string
+
+const (
+	SourceTypeGit  SourceType = "git"
+	SourceTypeHelm SourceType = "helm"
+)
+
+type Application struct {
```

Comment:
@jrryjcksn I saw a duplicate comment in the timeline here so I deleted it, and it appears to have removed BOTH comments. Here is what I was trying to say (from memory):

> If there are no objections, I can do that part in my stuff; I have to obtain those paths somewhere in order to do the manifest generation anyway.

I do have objections to bringing path writing stuff into `models.Application`. I would expect `WeGODirectoryWriter` to accept a `models.Application` as an argument and write it to the correct directory structure.

Sorry about the confusion there, I wasn't trying to delete my comment.

Reply:
I didn't want any path writing stuff in the model. I just wanted the model to be able to return what the paths are... The writing would still be separate. The paths are just derived from the model data. Also, it could be fine to move that into the manifest generation (which also needs to know what the paths are).

Reply:
@jrryjcksn 

>  I just wanted the model to be able to return what the paths are

To me, that is inherently "path stuff" and only relevant in a context where we are trying to write an `Application` to a repo. I have a feeling you and I mostly agree on this after our chat today.

Can I follow up with you outside of this PR so Justin can get this thing merged? If yes, I will mark this as resolved.

Reply:
Chatted with Jerry and we are :+1: